string mode # 'classic' or 'without_invalidation'
---
Prediction pred
